staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -161,15 +162,22 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		}
 
 		trace.Step("About to store object in database")
-		result, err := finishRequest(timeout, func() (runtime.Object, error) {
+		requestFunc := func() (runtime.Object, error) {
 			return r.Create(
 				ctx,
 				name,
 				obj,
 				rest.AdmissionToValidateObjectFunc(admit, admissionAttributes, scope),
 				options,
 			)
-		})
+		}
+		result, err := finishRequest(timeout, requestFunc)
+		if isTooLargeError(err) {
+			if accessor, accessorErr := meta.Accessor(obj); accessorErr == nil {
+				accessor.SetManagedFields(nil)
+				result, err = finishRequest(timeout, requestFunc)
```

Comment:
Do we double-call finishRequest in any other path? @deads2k @liggitt

Reply:
not that I know of